website/static/js/pages/support-page.js

```diff
@@ -0,0 +1,181 @@
+        }
+    }
+
+    /**
+     * Affix or restore search layer based on location
+     */
+    function fixSearchLayer () {
+        var topOffset = $(window).scrollTop();
+        var searchLayer = $('.search-layer');
+        if(topOffset > 100 && !searchLayer.hasClass('fixed-layer')){
+            searchLayer.addClass('fixed-layer');
+            $('.support-title').hide();
+            $('.search-up').removeClass('disabled');
+            $('.content-layer').addClass('content-padding'); // To avoid a suddent layout shift when affixing search layer
+        }
+        if(topOffset <= 100 && searchLayer.hasClass('fixed-layer')){
+            searchLayer.removeClass('fixed-layer');
+            $('.support-title').show();
+            $('.search-up').addClass('disabled');
+            $('.content-layer').removeClass('content-padding');
+        }
+    }
+
+    /* expand or collapse on clicking support item header */
+    $('.support-head').click(function(){
+        var item = $(this).parent();
+        changeExpandState(item, item.hasClass('open'));
+    });
+
+    /* Expand All button event  */
+    $('.search-expand').click(function(){
+        resetFilter();
+        $('.support-item').each(function(){
+            changeExpandState($(this));
+        });
+        updatePrevNextStatus();
+    });
+
+    /* Collapse All button event  */
+    $('.search-collapse').click(function(){
+        resetFilter();
+        $('.support-item').each(function(){
+            changeExpandState($(this), true);
+        });
+        updatePrevNextStatus();
+    });
+
+    /* Top button event to scroll to top*/
+    $('.search-up').click(function(){
+        scrollTo();
+    });
+
+    /* Previous button event to scroll view to the previous open support item */
+    $('.search-previous').click(function(){
+        if(searchItemIndex > 0){
+            var openItems = $('.support-item.open');
+            var prevEl = openItems.get(searchItemIndex-1);
+            scrollTo(prevEl);
+            $(prevEl).addClass('search-selected');
+            $(openItems.get(searchItemIndex)).removeClass('search-selected');
+            searchItemIndex--;
+        }
+        updatePrevNextStatus();
+    });
+
+    /* Next button event to scroll view to the next open support item */
+    $('.search-next').click(function(){
+        var openItems = $('.support-item.open');
+        var nextEl = openItems.get(searchItemIndex+1);
+        if(nextEl){
+            scrollTo(nextEl);
+            $(nextEl).addClass('search-selected');
+            $(openItems.get(searchItemIndex)).removeClass('search-selected');
+            searchItemIndex++;
+        }
+        updatePrevNextStatus();
+    });
+
+    $('.clear-search').click(resetFilter);
+
+    $('.support-filter').keyup(function(){
+        var text = $(this).val().toLowerCase();
+        if(text.length === 0){
+            resetFilter();
+        }
+        $('.clear-search').addClass('clear-active');
```

Comment:
When `text.length === 0`, `resetFilter()` will remove the active class from the "clear" button, but it immediately gets added back with this line. Do you need to `return` after `resetFilter()`?